hipcheck/src/plugin/mod.rs

```diff
@@ -17,3 +23,50 @@ pub fn dummy() {
 		/* jitter_percent */ 10,
 	);
 }
+
+pub async fn initialize_plugins(
+	plugins: Vec<(PluginContext, Value)>,
```

Comment:
I recommend making the tuple here into a struct, even just a tuple struct with public fields, so that there's a nameable type. In my experience in a context like this it'll make future discussions about the code and debugging of the code easier.

Reply:
Sure, can do. But I would think fixing `Value` to be more expressive (a la `type PluginConfiguration = serde_json::Value`) would be good enough. The `(Context, Config)` coupling is loose and ephemeral, so seeing visually that it's a pair that then gets used would be sufficient. Not to mention we'd have to have two structs to distinguish between `(Plugin, Conf)`, and `(PluginContext, Conf)`